### PR TITLE
todoist-electron: init at 1.19

### PR DESCRIPTION
**Pull request by i077**
`todoist-electron` is an Electron wrapper around the Todoist web app, with some global shortcuts.

Note: The source repository does not have a `COPYING` file to denote a license, so I am basing the license off of the [AUR Package](https://aur.archlinux.org/packages/todoist-electron) (ISC license), maintained by the owner of the source repo. Additionally, while the repo's name is `todoist-linux`, the AUR package name is `todoist-electron`, so I'm going with the latter since it's slightly more descriptive.

(this is also my first PR and first derivation for nixpkgs, so I may have gotten some things wrong)

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: maintainers/maintainer-list.nix [hunk 1]
```diff
@@ -2974,6 +2974,12 @@
     githubId = 12491746;
     name = "Masato Yonekawa";
   };
+  i077 = {
+    email = "contact@imranhossa.in";
+    github = "i077";
+    githubId = 2789926;
+    name = "Imran Hossain";
+  };
```
Comment (jonringer) [inline maintainers/maintainer-list.nix]: should be a separate commit
```
maintainers: add i077
```

File: pkgs/applications/misc/todoist-electron/default.nix [hunk 2]
```diff
@@ -0,0 +1,62 @@
+{ stdenv, lib, fetchurl, makeDesktopItem, dpkg, atk, at_spi2_atk, glib, pango, gdk-pixbuf
```
Comment (jonringer) [inline pkgs/applications/misc/todoist-electron/default.nix]:
```suggestion
{ stdenv, lib, fetchurl, makeDesktopItem, dpkg, atk, at_spi2_atk, glib, pango, gdk-pixbuf
```
for some reason, this can't find at_spi2_atk
```
error: while evaluating 'callPackagesWith' at /var/lib/ofborg/checkout/repo/38dca4e3aa6bca43ea96d2fcc04e8229/mr-est/hetzner-sb65-1083082-hel1-dc2/lib/customisation.nix:127:36, called from /var/lib/ofborg/checkout/repo/38dca4e3aa6bca43ea96d2fcc04e8229/mr-est/hetzner-sb65-1083082-hel1-dc2/pkgs/top-level/all-packages.nix:10686:22:
anonymous function at /var/lib/ofborg/checkout/repo/38dca4e3aa6bca43ea96d2fcc04e8229/mr-est/hetzner-sb65-1083082-hel1-dc2/pkgs/applications/misc/todoist-electron/default.nix:1:1 called without required argument 'at_spi2_atk', at /var/lib/ofborg/checkout/repo/38dca4e3aa6bca43ea96d2fcc04e8229/mr-est/hetzner-sb65-1083082-hel1-dc2/lib/customisation.nix:132:14
```

File: maintainers/maintainer-list.nix (hunk 1 above)
Comment (i077) [reply maintainers/maintainer-list.nix]: Got it. Just force pushed.

File: pkgs/applications/misc/todoist-electron/default.nix (hunk 2 above)
Comment (i077) [reply pkgs/applications/misc/todoist-electron/default.nix]: I'm not sure why, `nix-build` seems to work fine on my end. It produces an output at `/nix/store/gcpfk7bpj4j5z64vj0flz1dp9jk5g1gi-todoist-electron-1.19`.

Comment (i077) [reply pkgs/applications/misc/todoist-electron/default.nix]: Just realized I made a typo in `all-packages.nix`. Force pushed a fix.

Comment (jonringer) [reply pkgs/applications/misc/todoist-electron/default.nix]: I'm not sure either, I'll wait for the next ofborg eval

**Comment by jonringer** [comment]
@GrahamcOfBorg eval

**Comment by i077** [comment]
Ah, it's throwing an error because `at_spi2_atk` is an alias, and ofborg is failing on `allowAliases = false`. I'll force push a change to use the original name.

**Comment by i077** [comment]
@GrahamcOfBorg eval

**Comment by jonringer** [comment]
@GrahamcOfBorg eval

**Comment by jonringer** [comment]
ofborg has been slow recently, not sure why

**Comment by i077** [comment]
Yep, that fixed it. Everything's good to go.

File: pkgs/applications/misc/todoist-electron/default.nix [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+    # Patch binary
+    patchelf \
+      --set-interpreter "$(cat $NIX_CC/nix-support/dynamic-linker)" \
+      --set-rpath "${libPath}:\$ORIGIN" \
+      $out/opt/Todoist/todoist
+
+    # Hacky workaround for RPATH problems
+    makeWrapper $out/opt/Todoist/todoist $out/bin/todoist \
+      --prefix LD_LIBRARY_PATH : ${lib.makeLibraryPath [ libpulseaudio udev ]}
```
[the remark this answers is not in the document]
Comment (i077) [reply pkgs/applications/misc/todoist-electron/default.nix]: I actually had a problem where Nix would, while it was shrinking the RPATH, remove the library paths for `libpulseaudio` and `udev`, and thus the instance would crash upon startup. I had discussed this in the IRC and [was confused on why they got removed](https://logs.nix.samueldr.com/nixos/2020-02-03#3027834;), and couldn't figure out why Nix wanted to remove those paths.

Using a wrapper to export `LD_LIBRARY_PATH` before execution was the only solution I could think of.

[the remark this answers is not in the document]
Comment (i077) [reply pkgs/applications/misc/todoist-electron/default.nix]: [Here](https://pastebin.com/V9k6mkmQ) is the output when I build and print the RPATH of the ELF.

**Comment by jonringer** [comment]
@worldofpeace how do we handle top-level name conflicts. I feel like this should be called todoist, but there's already a go application by that name.

**Comment by i077** [comment]
I'm just going by how it's called in the AUR, I think it should be kept that way.

**Comment by jonringer** [comment]
I think todo-electron seems to be the most appropriate: https://repology.org/projects/?search=todoist&maintainer=&category=&inrepo=&notinrepo=&repos=&families=&repos_newest=&families_newest=

**Comment by i077** [comment]
You mean `todoist-electron`? If so, is this now ready to be merged in?

**Review by jonringer — APPROVED** [review]
commits LGTM
gui starts fine

```
[4 built, 0.0 MiB DL]
https://github.com/NixOS/nixpkgs/pull/79167
1 package built:
todoist-electron
```
